core/dbt/parser/schemas.py

```diff
@@ -817,23 +818,39 @@ def patch_constraints(self, node, constraints):
             node.constraints = [ModelLevelConstraint.from_dict(c) for c in constraints]
 
     def _validate_constraint_prerequisites(self, model_node: ModelNode):
+
+        column_warn_unsupported = [
+            constraint.warn_unsupported
+            for column in model_node.columns.values()
+            for constraint in column.constraints
+        ]
+        model_warn_unsupported = [
+            constraint.warn_unsupported for constraint in model_node.constraints
+        ]
+        warn_unsupported = column_warn_unsupported + model_warn_unsupported
+
+        # if any constraint has `warn_unsupported` as True then send the warning
+        if any(warn_unsupported) and model_node.config.materialized not in [
```

Comment:
could be refactored to use [materialization_enforces_constraints](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/contracts/graph/nodes.py#LL596C9-L596C45): `if any(warn_unsupported) and not model_node.materialization_enforces_constraints`

Reply:
Yes it 100% should be.